mods/env_sounds/init.lua

```diff
@@ -0,0 +1,121 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:getpos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = (num["default:water_flowing"] or 0) +
+		(num["default:river_water_source"] or 0) +
+		(num["default:river_water_flowing"] or 0)
+	-- If waters
+	if waters > 0 then
+		-- Find centre of water positions
+		local wposmid = wpos[1]
+		-- If more than 1 water
+		if #wpos > 1 then
+			local wposmin = areamax
+			local wposmax = areamin
+			for i = 1, #wpos do
+				local wposi = wpos[i]
+				if wposi.x > wposmax.x then
+					wposmax.x = wposi.x
+				end
+				if wposi.y > wposmax.y then
+					wposmax.y = wposi.y
+				end
+				if wposi.z > wposmax.z then
+					wposmax.z = wposi.z
+				end
+				if wposi.x < wposmin.x then
+					wposmin.x = wposi.x
+				end
+				if wposi.y < wposmin.y then
+					wposmin.y = wposi.y
+				end
+				if wposi.z < wposmin.z then
+					wposmin.z = wposi.z
+				end
+			end
+			wposmid = vector.divide(vector.add(wposmin, wposmax), 2)
```

Comment:
I think a weighted average calculation would be way shorter, faster and result in a much more accurate sound position:
```Lua
-- Starting at line 47
local pos_center = vector.new()
for i, pos in ipairs(wpos) do
	pos_center = vector.add(pos_center, pos)
end
pos_center = vector.divide(pos_center, #wpos)
-- Play sound
```

Reply:
Good point, will consider.

Reply:
Will do.